### PR TITLE
[WIP] Font

### LINKED ISSUE #138: Add attributed strings from libui-node 0.2.0
https://github.com/parro-it/libui-node/blob/master/docs/attributedstring.md

### PR DESCRIPTION
**Pull request by mischnic**
That `FancyText` (not sure about the name) isn't implemented yet
-> How can a React.Component determine the type of it's parents?

<br>

@kusti8 What should the default font family and size be (currently Arial 14pt)?

bildschirmfoto 2018-05-22 um 19 37 02

Closes #138

Some libui bugs:
- align: https://github.com/andlabs/libui/issues/368
- background on Windows: https://github.com/andlabs/libui/issues/347
- underline on macOS: https://github.com/andlabs/libui/issues/322

### CODE REVIEW

**Comment by kusti8** [comment]
Ok, its hard to get a component to know its parents. I sort of did something like that here https://github.com/kusti8/proton-native/blob/yoga/src/components/View.js, where I used the parent component and then basically chained function calls all the way up to the root. Default font should be Arial either 12pt or 14pt. Currently 12pt is pretty standard so that should be good.

I'll work on `FancyText`, (I'm thinking `StyledText`).

**Comment by mischnic** [comment]
> I'll work on `FancyText`, (I'm thinking `StyledText`).

👍 

> Anything else that needs to be done?

Everything from the libui api should be implemented (except for [OpenTypeFeatures](https://github.com/parro-it/libui-node/blob/1bc1b6b751b66bafbdd3ecf234bad7135ad07bf3/docs/attributedstring.md#opentypefeatures) which can for example be used to turn on/off ligatures).

**Comment by kusti8** [comment]
I added `StyledText` and I got the parent by using React Context. To do this, I had to upgrade `react-reconciler` so I had to make some changes so I need to test it more before merging.

**Comment by mischnic** [comment]
This needs to be adjusted to accept StyledText: (bottom of components/Area.js)
```
js
Area.Text.propTypes = {
  children: PropTypes.oneOfType([areaProp, PropTypes.arrayOf(areaProp)]),
};
```

**Comment by kusti8** [comment]
Looks like this is done. I'll merge if that's good with you @mischnic

File: src/components/Area.js [hunk 1]
```diff
@@ -734,4 +738,168 @@ Area.Path.defaultProps = {
...
+          textAlign = libui.textAlign.center;
+          break;
+        case 'right':
+          textAlign = libui.textAlign.right;
+          break;
+      }
+
+      const font = new libui.FontDescriptor(
+        'Arial',
+        14,
```
Comment (mischnic) [inline src/components/Area.js]: > Looks like this is done. I'll merge if that's good with you @mischnic

👍 

You might want to change this to `12` for the default font (that's also the default font size of the font button on macOS).

**Pull request merged**